app/domain/authentication/authn_jwt/identity_providers/create_identity_provider.rb

```diff
@@ -0,0 +1,62 @@
+require 'command_class'
+
+module Authentication
+  module AuthnJwt
+    module IdentityProviders
+      # Factory for jwt identity providers.
+      # If Identity variable is configured factory return the decoded_token_provider
+      # If the Identity variable is not configured and there is account field in url factory returns url provider
+      # If the above conditions are not met exception is raised
+      CreateIdentityProvider = CommandClass.new(
+        dependencies: {
+          identity_from_url_provider_class: Authentication::AuthnJwt::IdentityProviders::IdentityFromUrlProvider,
+          identity_from_decoded_token_class: Authentication::AuthnJwt::IdentityProviders::IdentityFromDecodedTokenProvider,
+          logger: Rails.logger
+        },
+        inputs: %i[authentication_parameters]
+      ) do
+        def call
+          create_identity_provider
+        end
+
+        private
+
+        def identity_from_decoded_token_provider
+          @identity_from_decoded_token_provider ||= @identity_from_decoded_token_class.new(@authentication_parameters)
+        end
+
+        def identity_from_url_provider
+          @identity_from_url_provider ||= @identity_from_url_provider_class.new(@authentication_parameters)
+        end
+
+        def create_identity_provider
+          @logger.debug(LogMessages::Authentication::AuthnJwt::SelectingIdentityProviderInterface.new)
+
+          check_identity_configuration
+          if identity_from_decoded_token_provider.identity_available
+            @logger.info(
+              LogMessages::Authentication::AuthnJwt::SelectedIdentityProviderInterface.new(
+                TOKEN_IDENTITY_PROVIDER_INTERFACE_NAME
+              )
+            )
+            return identity_from_decoded_token_provider
+          elsif identity_from_url_provider.identity_available
+            @logger.info(
+              LogMessages::Authentication::AuthnJwt::SelectedIdentityProviderInterface.new(
+                URL_IDENTITY_PROVIDER_INTERFACE_NAME
+              )
+            )
+            return identity_from_url_provider
+          end
+        end
+
+        def check_identity_configuration
+          if (identity_from_decoded_token_provider.identity_available and identity_from_url_provider.identity_available) ||
+            (!identity_from_decoded_token_provider.identity_available and !identity_from_url_provider.identity_available)
+            raise Errors::Authentication::AuthnJwt::IdentityMisconfigured
```

Comment:
do we want to raise specific errors for each case?

Reply:
Yes

Reply:
We have troubleshooting section for this

Reply:
well here you don't raise errors for specific cases. you raise the same error for cases where the customer configured both options and where the customer didn't configure any

Reply:
Yes i raise same error for both cases. Similar like siging key. There is addressed in documentation troubleshooting

Reply:
np, just raised a UX concern. i think that we should raise different errors so that it is clear from the logs and the user doesn't need to go to the troubleshooting section in the docs. your call.
cc: @hilagross

Reply:
I agree with @orenbm, but as I'm currently discuss with PM about if we need to keep the URL option or not we should wait on that.